esp-bootloader-esp-idf/src/partitions.rs

```diff
@@ -493,11 +491,19 @@ impl<F> FlashRegion<'_, F> {
     pub fn partition_size(&self) -> usize {
         self.raw.len() as _
     }
+
+    fn range(&self) -> core::ops::Range<u32> {
+        self.raw.offset()..self.raw.offset() + self.raw.len()
+    }
+
+    fn in_range(&self, start: u32, len: usize) -> bool {
+        self.range().contains(&start) && self.range().contains(&(start + len as u32 - 1))
```

Comment:
```suggestion
        self.range().contains(&start) && (start + len as u32 <= self.range().end)
```

Reply:
looking forward to the new Range API getting stabilized one day :)